frontend/src/app/patients/Components/FacilitySelect.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { useSelector } from "react-redux";
+
+import Select from "../../commonComponents/Select";
+
+interface Props {
+  facilityId: string | null;
+  onChange: (facilityId: string | null) => void;
+  validateField: () => void;
+  validationStatus: (name: keyof PersonFormData) => "error" | undefined;
+  errors: { [key: string]: string | undefined };
+  hidden?: boolean;
+}
+
+const ALL_FACILITIES = "~~ALL-FACILITIES~~";
+const NAME = "facilityId";
+
+const FacilitySelect: React.FC<Props> = (props) => {
+  const facilities = useSelector(
+    (state) => (state as any).facilities as Facility[]
+  );
+
+  if (props.hidden) {
+    return null;
+  }
+
+  const facilityList = facilities.map((f: any) => ({
+    label: f.name,
+    value: f.id,
+  }));
+  facilityList.unshift({ label: "All facilities", value: ALL_FACILITIES });
+  facilityList.unshift({ label: "-Select-", value: "" });
```

Comment:
It appears that these were added because the `defaultSelect` has `value={undefined}` which then equates the `input.value === "- Select -"` instead of `""`.


Updating this and the default option